get-resource.sh

```diff
@@ -41,8 +41,19 @@ if [ "$FILECACHED" == "${RHCOS_IMAGE_FILENAME_OPENSTACK}" ] ; then
     curl -O "$CACHEURL/$RHCOS_IMAGE_FILENAME_OPENSTACK/$RHCOS_IMAGE_FILENAME_OPENSTACK"
     curl -O "$CACHEURL/$RHCOS_IMAGE_FILENAME_OPENSTACK/$RHCOS_IMAGE_FILENAME_COMPRESSED.md5sum"
 else
-    curl --insecure --compressed -L --dump-header "${RHCOS_IMAGE_FILENAME_OPENSTACK}.headers" -o "${RHCOS_IMAGE_FILENAME_OPENSTACK}" "${RHCOS_IMAGE_URL}/${RHCOS_IMAGE_FILENAME_OPENSTACK}"
-    qemu-img convert -O qcow2 -c "$RHCOS_IMAGE_FILENAME_OPENSTACK" "$RHCOS_IMAGE_FILENAME_COMPRESSED"
+    # Check for a cached copy of the file on the host in the VM/bootstrap case
+    if virt-what | grep -q kvm; then
+        DEFAULT_GW=$(ip r | grep default | grep dhcp | awk '{print $3}')
+        if curl -I http://$DEFAULT_GW/images/$RHCOS_IMAGE_FILENAME_COMPRESSED; then
+            curl -o "$RHCOS_IMAGE_FILENAME_COMPRESSED" http://$DEFAULT_GW/images/$RHCOS_IMAGE_FILENAME_COMPRESSED
```

Comment:
Can we grab the headers file aswell, to avoid hitting this logic is the container is run twice and we don't need to.

Reply:
Ah I'm only downloading the compressed version to avoid re-running qemu-img, but I'll add a test so we download the ```"${RHCOS_IMAGE_FILENAME_OPENSTACK}.headers"``` file from the host as well, and that should exist provided the downloader container is used to create the cache on the host.